pallets/automation-time/src/lib.rs

```diff
@@ -1094,25 +1103,27 @@ pub mod pallet {
 
 		pub fn run_xcmp_task(
 			para_id: ParaId,
-			call: Vec<u8>,
-			weight_at_most: Weight,
+			caller: T::AccountId,
+			currency_id: T::CurrencyId,
+			encoded_call: Vec<u8>,
+			encoded_call_weight: Weight,
 			task_id: T::Hash,
 		) -> Weight {
-			let destination = (1, Junction::Parachain(para_id.into()));
-			let message = Xcm(vec![Transact {
-				origin_type: OriginKind::Native,
-				require_weight_at_most: weight_at_most,
-				call: call.into(),
-			}]);
-			match T::XcmSender::send_xcm(destination, message) {
+			match T::XcmpTransactor::transact_xcm(
+				para_id.into(),
+				currency_id,
+				caller,
+				encoded_call,
+				encoded_call_weight,
+			) {
 				Ok(()) => {
-					Self::deposit_event(Event::SuccessfullySentXCMP { task_id, para_id });
+					Self::deposit_event(Event::XcmpTaskSucceeded { task_id, para_id });
 				},
 				Err(e) => {
-					Self::deposit_event(Event::FailedToSendXCMP { task_id, para_id, error: e });
+					Self::deposit_event(Event::XcmpTaskFailed { task_id, para_id, error: e });
 				},
 			}
-			// Adding 1 DB write that doesn't get accounted for in the benchmarks to run an xcmp task
```

Comment:
Adding this line back.

Reply:
you lost me

Reply:
I didn't mean to remove this line as it's still relevant.

Reply:
updated line to `2 writes and 1 read`.